airbyte/_executors/python.py

```diff
@@ -193,6 +245,40 @@ def get_installed_version(
 
             return None
 
+    def _check_python_version_compatibility(
+        self,
+        package_name: str,
+        requires_python: str | None,
+    ) -> bool | None:
```

Comment:
This seems like a waste of a class method. Move to a util module, maybe in a new '_semver' submodule, and then simplify by calling the semver helper upstream. Doesn't seem like it belongs here.